types/src/transaction/authenticator.rs

```diff
@@ -95,6 +99,11 @@ impl TransactionAuthenticator {
                 secondary_signer_addresses,
                 secondary_signers,
             } => {
+                let num_sigs: usize = sender.number_of_signatures() +
```

Comment:
One idea: we could extract this `number_signatures` logic to a separate function, then check `num_sigs > MAX_NUM_OF_SIGS` at the beginning of `verify`. That would make the intention a bit clearer: signature limit is a concept that applies to all authenticators, not just multi-agent. This would also give us some additional protection in the case that the multi-ed limit check is removed or relaxed.

Reply:
I've moved it to the beginning of `verify`.